torax/_src/pedestal_model/register_model.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2025 DeepMind Technologies Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Register a pedestal model with TORAX."""
+from typing import Union, get_args
+
+from torax._src.torax_pydantic import model_config
+from torax._src.pedestal_model import pydantic_model
+
+
+def register_pedestal_model(
+    pydantic_model_class: type[pydantic_model.BasePedestal],
+):
+  """Registers a pedestal model with TORAX.
+
+  This function adds the pedestal model to the config model such that it can
+  be configured via pydantic. The pydantic model class should inherit from
+  BasePedestal and should have a distinct model_name. It should also define a
+  build_pedestal_model method which returns a PedestalModel.
+
+  Example:
+    ```python
+    from torax._src.pedestal_model import pydantic_model
```

Comment:
Make a public pedestal API like we already have for transport models to use in the example